From 4e4452b3bba71265864d9c46f135418ffd6113a0 Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Wed, 9 Nov 2022 09:07:21 -0300
Subject: [PATCH] STX: Customize Debian build files

Adds to libvirt Debian build instructions, starlingx specific files
based on how starlingx-staging/stx-libvirt was previously built and
delivered on CentOS, adding to the following:
  * Enabling syslog for libvirtd
  * Setting "auth_tcp" to "none"
  * Logrotate files
  * Qemu hook

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
---
 debian/libvirt-daemon-system.install |  1 +
 debian/rules                         | 24 ++++++++++++++++++++++++
 2 files changed, 25 insertions(+)

diff --git a/debian/libvirt-daemon-system.install b/debian/libvirt-daemon-system.install
index 04cf9052c..b2ce2ac75 100644
--- a/debian/libvirt-daemon-system.install
+++ b/debian/libvirt-daemon-system.install
@@ -2,6 +2,7 @@ etc/default/libvirt-guests
 etc/default/libvirtd
 etc/default/virtlockd
 etc/default/virtlogd
+etc/libvirt/hooks/qemu
 etc/libvirt/libvirtd.conf
 etc/libvirt/qemu-lockd.conf
 etc/libvirt/qemu.conf
diff --git a/debian/rules b/debian/rules
index a495870c4..a34132779 100755
--- a/debian/rules
+++ b/debian/rules
@@ -15,6 +15,11 @@ DEB_LDFLAGS_MAINT_STRIP = -Wl,-Bsymbolic-functions
 DPKG_EXPORT_BUILDFLAGS = 1
 include /usr/share/dpkg/buildflags.mk
 
+SOURCE1                   = libvirt.logrotate
+SOURCE2                   = libvirt.lxc
+SOURCE3                   = libvirt.qemu
+SOURCE4                   = qemu
+
 ARCHES_LXC  = alpha amd64 arm64 armel armhf hppa i386 m68k mips64el mipsel powerpc ppc64 ppc64el riscv64 s390x sh4 sparc64 x32
 ARCHES_XEN  = amd64 arm64 armhf i386
 ARCHES_VBOX = amd64 i386
@@ -221,6 +226,25 @@ override_dh_auto_install:
 	   $(DEB_DESTDIR)/etc/libvirt/nwfilter/ \
 	   $(DEB_DESTDIR)/usr/share/libvirt/
 
+	# STX: Begin custom install
+	# Enable syslog for libvirtd (/var/log/libvirtd.log)
+	echo "log_outputs=\"3:syslog:libvirtd\"" >> $(DEB_DESTDIR)/etc/libvirt/libvirtd.conf
+
+	# Set auth_tcp to "none" for now to enable live migration.
+	# We'll need to set up proper authentication later.
+	sed -i '/#auth_tcp/a auth_tcp = "none"' $(DEB_DESTDIR)/etc/libvirt/libvirtd.conf
+
+	# Install logrotate files
+	install -p -D -m 644 $(SOURCE1) $(DEB_DESTDIR)/etc/logrotate.d/libvirtd
+	install -p -D -m 644 $(SOURCE2) $(DEB_DESTDIR)/etc/logrotate.d/libvirtd.lxc
+	install -p -D -m 644 $(SOURCE3) $(DEB_DESTDIR)/etc/logrotate.d/libvirtd.qemu
+
+	# Install hooks
+	mkdir -p $(DEB_DESTDIR)/etc/libvirt/hooks
+	install -m 0500 $(SOURCE4) $(DEB_DESTDIR)/etc/libvirt/hooks/qemu
+
+	# STX: End custom install
+
 override_dh_install-arch:
 	dh_install
 
-- 
2.25.1

